### PR TITLE
NoteSync hybrid sample app and SFContentSoqlSyncTarget custom target example

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

**Comment by khawkins** [comment]
Should we piggyback on the assets in shared/resources/Images.xcassets?  I realize that there are a couple of other hybrid sample apps that haven't been cut over yet, but seems like we should be going in that direction.

**Comment by khawkins** [comment]
I don't know if it's related to this PR, but the Connected App ID from shared doesn't seem to be the sample Connected App that we generally use.  Doesn't work against production.

**Comment by khawkins** [comment]
Getting an error in post-authentication:

        2015-02-20 13:54:52.167 NoteSync[65222:1221629] Error in success callback: com.salesforce.oauth25645269 = TypeError: undefined is not a constructor (evaluating 'new app.Router()')

Seems like maybe there's a order of precedence issue in NoteSync.html, perhaps?

File: hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+// Params for SFSoapSoqlRequest
+#define SESSION_ID @"sessionId"
+#define QUERY @"query"
+#define IS_LOCATOR @"query"
+
+typedef void (^SFSoapSoqlResponseParseComplete) ();
+
+@interface SFRestAPI ()
+
+- (void)removeActiveRequestObject:(SFRestRequest *)request;
```
Comment (khawkins) [inline hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m]: Should we not expose this method?  Or do you think it's a low enough percentage use case to leave it as is?

File: hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+-(void)parse:(NSData*)data completeBlock:(SFSoapSoqlResponseParseComplete)completeBlock
+{
+    self.completeBlock = completeBlock;
+    self.xmlParser = [[NSXMLParser alloc] initWithData:data];
+    self.xmlParser.delegate = self;
+    [self.xmlParser parse];
+}
+
+-(void)parserDidStartDocument:(NSXMLParser *)parser
```
Comment (khawkins) [inline hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m]: Whee!  SOAP!

File: hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    
+    self.networkOperation.delegate = self;
+    [networkEngine enqueueOperation:self.networkOperation];
+    
+    return self.networkOperation;
+}
+
+- (void)networkOperationDidFinish:(SFNetworkOperation *)networkOperation {
+    __weak SFSoapSoqlRequest* weakSelf = self;
+    if ([self.delegate respondsToSelector:@selector(request:didLoadResponse:)]) {
```
Comment (khawkins) [inline hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m]: We're not actually using `weakSelf` in here.

Comment (khawkins) [inline hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m]: Or not in all cases, anyway.

File: hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+        self.endpoint = SOAP_ENDPOINT;
+        self.parseResponse = NO;
+        self.query = nil;
+        self.queryLocator = queryLocator;
+    }
+    return self;
+}
+
+
+- (SFNetworkOperation*) send:(SFNetworkEngine*) networkEngine {
```
Comment (khawkins) [inline hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m]: We're going to need to redo this exercise shortly with the new Network SDK.

File: libs/SalesforceRestAPI/SalesforceRestAPI/Classes/SFRestAPI.m [hunk 5]
```diff
@@ -31,7 +31,7 @@
 #import <SalesforceSDKCore/SFSDKWebUtils.h>
 #import <SalesforceSDKCore/SalesforceSDKManager.h>
 
-NSString* const kSFRestDefaultAPIVersion = @"v31.0";
+NSString* const kSFRestDefaultAPIVersion = @"v32.0";
```
Comment (wmathurin) [inline libs/SalesforceRestAPI/SalesforceRestAPI/Classes/SFRestAPI.m]: ContentNote are in 32 (anyway we should always point to the most recent release number)

Comment (khawkins) [inline libs/SalesforceRestAPI/SalesforceRestAPI/Classes/SFRestAPI.m]: :+1:

**Comment by khawkins** [comment]
Looks fine, over all.

File: libs/SalesforceRestAPI/SalesforceRestAPI/Classes/SFRestAPI.m [hunk 6]
```diff
@@ -31,7 +31,7 @@
 #import <SalesforceSDKCore/SFSDKWebUtils.h>
 #import <SalesforceSDKCore/SalesforceSDKManager.h>
 
-NSString* const kSFRestDefaultAPIVersion = @"v31.0";
+NSString* const kSFRestDefaultAPIVersion = @"v33.0";
```
Comment (wmathurin) [inline libs/SalesforceRestAPI/SalesforceRestAPI/Classes/SFRestAPI.m]: 33 is the most recent api (spring 15)

**Comment by bhariharan** [comment]
The connected app consumer key is ```Mobile2```. It needs to be switched back to ```Sandbox``` at some point in all 3 repos.

**Comment by bhariharan** [comment]
:+1:

File: hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m (hunk 1 above)
Comment (wmathurin) [inline hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m]: It's only useful if you need to provide your own implementation for send. I had to do that in order to be able to send xml data.

File: hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m (hunk 4 above)
Comment (wmathurin) [inline hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m]: Maybe with network sdk we will be able to pass any http entity for the body (like on Android)?

Comment (khawkins) [inline hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m]: If we can't, then we'll refactor it so we can.

**Comment by wmathurin** [comment]
Regarding the error in post-authentication, I haven't seen it myself, but will keep an eye out for it.
The fix if any will go into the Shared repo so merging this pull request.

**Pull request merged**